Source/JavaScriptCore/heap/FreeListInlines.h

```diff
@@ -33,35 +33,45 @@ namespace JSC {
 template<typename Func>
 ALWAYS_INLINE HeapCell* FreeList::allocate(const Func& slowPath)
 {
-    unsigned remaining = m_remaining;
-    if (remaining) {
-        unsigned cellSize = m_cellSize;
-        remaining -= cellSize;
-        m_remaining = remaining;
-        return bitwise_cast<HeapCell*>(m_payloadEnd - remaining - cellSize);
+    unsigned cellSize = m_cellSize;
+    if (LIKELY(m_payloadStart != m_payloadEnd)) {
+        char* result = m_payloadStart;
+        m_payloadStart += cellSize;
+        return bitwise_cast<HeapCell*>(result);
     }
     
-    FreeCell* result = head();
-    if (UNLIKELY(!result))
+    FreeCell* cell = head();
+    if (isSentinel(cell))
```

Comment:
UNLIKELY?

Reply:
Added this back in latest push, this should be pretty unlikely.

Reply:
Added this back in latest push, this should be pretty unlikely.

Reply:
Added this back in latest push, this should be pretty unlikely.